GPU Decord and Tensorflow bridge

I am working with GPU accelerated Decord and am running into an issue using Decord with Tensorflow. In particular, I am running into the error below

```
Traceback (most recent call last):
  File "src/dataloader.py", line 373, in <module>
    frames = vr[0:10]
  File "/root/.local/lib/python3.6/site-packages/decord-0.4.2-py3.6-linux-x86_64.egg/decord/video_reader.py", line 96, in __getitem__
    return self.get_batch(range(*idx.indices(len(self))))
  File "/root/.local/lib/python3.6/site-packages/decord-0.4.2-py3.6-linux-x86_64.egg/decord/video_reader.py", line 173, in get_batch
    return bridge_out(arr)
  File "/root/.local/lib/python3.6/site-packages/decord-0.4.2-py3.6-linux-x86_64.egg/decord/bridge/__init__.py", line 40, in bridge_out
    return _BRIDGE_TYPES[_CURRENT_BRIDGE.type][0](native_arr)
  File "/root/.local/lib/python3.6/site-packages/decord-0.4.2-py3.6-linux-x86_64.egg/decord/bridge/tf.py", line 21, in to_tensorflow
    return tfdl.from_dlpack(decord_arr.to_dlpack())
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/dlpack/dlpack.py", line 66, in from_dlpack
    return pywrap_tfe.TFE_FromDlpackCapsule(dlcapsule, context.context()._handle)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/context.py", line 752, in _handle
    raise AssertionError("Context must be initialized first.")
AssertionError: Context must be initialized first.
```

The code that produces this error is:

```
vr = VideoReader(video_path, ctx=gpu(0))
decord.bridge.set_bridge('tensorflow')
frames = vr[0:10]
print(frames.shape)
```

I can tell it is the Tensorflow bridge causing issues as if I remove the second line everything works fine and frames are correctly extracted. Also if I switch the context to cpu then the code works as well no problem. Would appreciate any help here.

This is happening with Tensorflow v2.3.0, Python 3.6.9, and CUDA 10.1.

--- a/python/decord/bridge/tf.py
+++ b/python/decord/bridge/tf.py
@@ -18,7 +18,11 @@ def try_import_tfdl():
 def to_tensorflow(decord_arr):
     """from decord to tensorflow, no copy"""
     tfdl = try_import_tfdl()
-    return tfdl.from_dlpack(decord_arr.to_dlpack())
+    from tensorflow.python import pywrap_tfe
+    from tensorflow.python.eager import context
+    ctx = context.context()
+    ctx.ensure_initialized()
+    return pywrap_tfe.TFE_FromDlpackCapsule(decord_arr.to_dlpack(), ctx._handle)
 
 def from_tensorflow(tf_tensor):
     """from tensorflow to decord, no copy"""

--- a/python/decord/video_reader.py
+++ b/python/decord/video_reader.py
@@ -62,8 +62,11 @@ class VideoReader(object):
         self._avg_fps = None
 
     def __del__(self):
-        if self._handle:
-            _CAPI_VideoReaderFree(self._handle)
+        try:
+            if self._handle is not None:
+                _CAPI_VideoReaderFree(self._handle)
+        except TypeError:
+            pass
 
     def __len__(self):
         """Get length of the video. Note that sometimes FFMPEG reports inaccurate number of frames,

